pkg/middleware/auth.go

```diff
@@ -90,6 +91,52 @@ func (c JWTVerifier) CheckToken(token string) (*jwt.Token, error) {
 	return tk, nil
 }
 
+// JWTVerifierURL checks token with JWT against JWKS, implements Verifier
+type JWTVerifierURL struct {
+	jwksURL string
+
+	parser *jwt.Parser
+}
+
+// NewJWTVerifierURL creates JWTVerifierURL with JWKS URL
+func NewJWTVerifierURL(jwksURL string) JWTVerifierURL {
+	return JWTVerifierURL{jwksURL: jwksURL, parser: new(jwt.Parser)}
+}
+
+// CheckToken checks the token, validates against JWKS and returns it if it's valid
+func (c JWTVerifierURL) CheckToken(token string) (*jwt.Token, error) {
+	if token == "" {
+		return nil, errors.New("invalid token")
```

Comment:
Errors returned from `CheckToken` end up rendered into the response, and the response status is 401. Are you sure all these errors should be displayed to the user, and should be 401? For example I think if the fetch fails, perhaps 500 or 503 is better. This comment also applies to the `CheckToken` method of `JWTVerifier`.

Reply:
@mjc1283 and I talked about it, 401 is fine, we will just change the message to "empty token"

Reply:
Agreed that the code should reflect user vs server error.